### PR TITLE
feat: ELB & certs for seed addresses

### PR DESCRIPTION
**Pull request by ktechmidas**
## Issue being fixed or feature implemented
The feature here is the creation and configuration of an AWS Elastic Load Balancer (ELB), ACM Certificate, Route53 Record and Security Group using Terraform. This setup is required to make seed-*.testnet.networks.dash.org accessible over HTTPS since the old method will not work while DAPI has differing certificates.

## What was done?
New ELB for seeds with the correct certificates, adding HPMNs as backends on port 3001. 

## How Has This Been Tested?
devnet-dude

## Breaking Changes
N/A

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: terraform/aws/main.tf [hunk 1]
```diff
@@ -148,6 +148,74 @@ resource "aws_elb" "web" {
...
+  internal           = false
+  load_balancer_type = "application"
+  security_groups    = [aws_security_group.sg-seed.id]
+  subnets            = aws_subnet.public.*.id
+  enable_deletion_protection = false
+}
+
+resource "aws_lb_listener" "listener" {
+  load_balancer_arn = aws_lb.seed-elb.arn
+  port              = 443
```
Comment (shumkov) [inline terraform/aws/main.tf:162]: It should be dapi_port variable

File: terraform/aws/main.tf [hunk 2]
```diff
@@ -148,6 +148,74 @@ resource "aws_elb" "web" {
...
+
+  default_action {
+    type             = "forward"
+    target_group_arn = aws_lb_target_group.tg-seed.arn
+  }
+}
+
+resource "aws_lb_target_group" "tg-seed" {
+  name     = "${var.public_network_name}-tg-seed"
+  port     = 3001
```
Comment (shumkov) [inline terraform/aws/main.tf:175]: It should be dapi_port as well

File: terraform/aws/security_groups.tf [hunk 3]
```diff
@@ -444,3 +444,28 @@ resource "aws_security_group" "vpn" {
...
 }
+
+resource "aws_security_group" "sg-seed" {
+  name        = "seed-1"
+  description = "Allow inbound traffic on port 443 and all outbound traffic"
+  vpc_id      = aws_vpc.default.id
+
+  ingress {
+    description = "TLS from VPC"
+    from_port   = 443
```
Comment (shumkov) [inline terraform/aws/security_groups.tf:455]: dapi_port?

**Review by shumkov — DISMISSED** [review]

File: terraform/aws/main.tf [hunk 4]
```diff
@@ -148,6 +148,74 @@ resource "aws_elb" "web" {
   }
 }
 
+resource "aws_lb" "seed-elb" {
```
Comment (strophy) [inline terraform/aws/main.tf:151]: Snake case for variable names instead of kebab case

File: terraform/aws/security_groups.tf [hunk 5]
```diff
@@ -444,3 +444,28 @@ resource "aws_security_group" "vpn" {
     DashNetwork = terraform.workspace
   }
 }
+
+resource "aws_security_group" "sg-seed" {
```
Comment (strophy) [inline terraform/aws/security_groups.tf:448]: Other security groups are not named with `sg-` prefix, follow naming convention

File: terraform/aws/main.tf [hunk 6]
```diff
@@ -256,14 +324,47 @@ resource "random_shuffle" "dns_ips" {
...
 
 # `seed-n` type addresses are dapi endpoints
-resource "aws_route53_record" "masternodes" {
-  zone_id = data.aws_route53_zone.main_domain[0].zone_id
-  name    = "seed-${count.index + 1}.${var.public_network_name}.${var.main_domain}"
-  type    = "A"
-  ttl     = "300"
-  records = random_shuffle.dns_ips.result
 
-  count = length(var.main_domain) > 1 ? 5 : 0
```
Comment (strophy) [inline terraform/aws/main.tf:266]: Can we keep this method of iteration? The new code is less flexible in the number of R53 records created.

**Review by strophy — CHANGES_REQUESTED** [review]
Nice work, general design looks good. Left a few code comments. Would a network load balancer be sufficient here, why do we need the ALB?

**Comment by ktechmidas** [comment]
@strophy could have gone either way, but if DAPI could, for example, tell us if a node is banned it could be removed, A/B testing could be done etc

An ALB is only a tiny bit more expensive and gives us more options

plus:

> Both Classic Load Balancers and Application Load Balancers use connection multiplexing, but Network Load Balancers do not. Therefore, your targets can receive more TCP connections behind a Network Load Balancer. Be sure that your targets are prepared to handle the volume of connection requests they might receive.

File: terraform/aws/security_groups.tf [hunk 7]
```diff
@@ -444,3 +444,28 @@ resource "aws_security_group" "vpn" {
...
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
```
Comment (strophy) [inline terraform/aws/security_groups.tf:466]: What is this for?

File: terraform/aws/security_groups.tf [hunk 8]
```diff
@@ -444,3 +444,28 @@ resource "aws_security_group" "vpn" {
...
   }
 }
+
+resource "aws_security_group" "seed" {
+  name        = "seed-1"
+  description = "Allow inbound traffic on DAPI port and all outbound traffic"
+  vpc_id      = aws_vpc.default.id
+
+  ingress {
+    description = "TLS from VPC"
```
Comment (strophy) [inline terraform/aws/security_groups.tf:454]: Isn't this from everywhere?

**Review by strophy — APPROVED** [review]
Nice work mate!

**Pull request merged**